web/src/pm_list.ts

```diff
@@ -57,22 +61,22 @@ export function _build_direct_messages_list() {
     return dom_ast;
 }
 
-function set_dom_to(new_dom) {
+function set_dom_to(new_dom: vdom.Tag<PMNode>): void {
     const $container = scroll_util.get_content_element($("#direct-messages-list"));
 
-    function replace_content(html) {
-        $container.html(html);
+    function replace_content(html: string | undefined): void {
```

Comment:
I'm not sure it makes sense to pass `undefined` to this function.

Reply:
It's a required type for `vdom.update`

Reply:
OK but that looks like a workaround for the types being wrong in `vdom.ts`; this can't actually be undefined.

```   
 if (opts.keyed_nodes === undefined) {                                                               
        blueslip.error("We need keyed_nodes to render innards.");                                       
        return undefined;                                                                               
    }                                                                                                   

```